src/Microsoft.ML.Data/Evaluators/BinaryClassifierEvaluator.cs

```diff
@@ -1097,18 +1097,18 @@ private void CheckInputColumnTypes(Schema schema)
 
             var t = schema[(int) LabelIndex].Type;
             if (t != NumberType.R4 && t != NumberType.R8 && t != BoolType.Instance && t.GetKeyCount() != 2)
-                throw Host.Except("Label column '{0}' has type '{1}' but must be R4, R8, BL or a 2-value key", LabelCol, t);
```

Comment:
>'{1}' but must be R4, R8, BL [](start = 63, length = 28)

some of them don't apply anymore for ML.Net, because for the estimators we create the label column to be of a specific type, depending on the task.. I think only bool applies here; BUT the catch is that the message would be incorrect for the CDM users, if we trim it to just bool. 

@tomfinley, @glebuk for preferences.

Reply:
My preference would be for the type description to be accurate w.r.t. the actual requirements. From the condition, it is clearly allowing for float and double inputs, so the message ought to reflect that.

If we wanted to form an actual restriction that this *must* be a `bool`, I would treat that as a separate discussion to improving the actual messages. There might be a case to be made for that. (Indeed, in the so-called "static" API, I made this a requirement.) But I'd want to discuss that issue separately. Right now I'd say it's most important that the message be correct.